### PR TITLE
Standardise dataproc location param to region

### LINKED ISSUE #15622: Inconsistencies with Dataproc Operator parameters
I'm looking at the GCP Dataproc operator and noticed that the `DataprocCreateClusterOperator` and `DataprocDeleteClusterOperator` require a `region` parameter, but other operators, like the `DataprocSubmitJobOperator` require a `location` parameter instead. I think it would be best to consistently enforce the parameter as `region`, because that's what is required in the protos for all of the [cluster CRUD operations](https://github.com/googleapis/python-dataproc/blob/master/google/cloud/dataproc_v1/proto/clusters.proto) and for [job submission](https://github.com/googleapis/python-dataproc/blob/d4b299216ad833f68ad63866dbdb2c8f2755c6b4/google/cloud/dataproc_v1/proto/jobs.proto#L741). `location` also feels too ambiguous imho, because it implies we could also pass a GCE zone, which in this case is either unnecessary or not supported (I can't remember which and it's too late in my Friday for me to double check 🙃 )

This might be similar to #13454 but I'm not 100% sure. If we think this is worth working on, I could maybe take this on as a PR, but it would be low priority for me, and if someone else wants to take it on, they should feel free to 😄  

IMPORTANT!!!

PLEASE CHECK "SIMILAR TO X EXISTING ISSUES" OPTION IF VISIBLE
NEXT TO "SUBMIT NEW ISSUE" BUTTON!!!

PLEASE CHECK IF THIS ISSUE HAS BEEN REPORTED PREVIOUSLY USING SEARCH!!!

Please complete the next sections or the issue will be closed.
These questions are the first thing we need to know to understand the context.

-->

**Apache Airflow version**: 2.0.1

**Environment**:

running locally on a Mac with SQLite backend only running a unit test that makes sure the dag is compiled. Installed with Pip and the provided constraints file. 

**What happened**:

if you pass `location` in the `DataprocCreateClusterOperator` the DAG won't compile and will throw an error `airflow.exceptions.AirflowException: Argument ['region']`

As minimally and precisely as possible. Keep in mind we do not have access to your cluster or dags.

--->

### PR DESCRIPTION
**Pull request by Daniel-Han-Yang**
Standardises DataProc hook & operators `location` parameter to `region` in line
with underlying google DataProc Python client library.

closes: #15622

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by mik-laj** [comment]
It looks like a breaking change. Can we avoid it and keep backward compatibility with warning? This change also requires an entry in changelog to facilitate migration. See: https://github.com/apache/airflow/blob/master/airflow/providers/google/CHANGELOG.rst

**Comment by leahecole** [comment]
> It looks like a breaking change. Can we avoid it and keep backward compatibility with warning? This change also requires an entry in changelog to facilitate migration. See: https://github.com/apache/airflow/blob/master/airflow/providers/google/CHANGELOG.rst

Thank you so much for doing this change that I haven't had time to do! +1 to @mik-laj 's comment about it being a breaking change.

**Comment by Daniel-Han-Yang** [comment]
@mik-laj Would this now be alright? Please let me know

**Review by mik-laj — APPROVED** [review]

File: airflow/providers/google/CHANGELOG.rst [hunk 1]
```diff
@@ -15,6 +15,16 @@
...
+Changelog
+---------
+
+4.1.0
+.....
+
+Bug Fixes
+~~~~~~~~~
+* ``Fix: Inconsistencies with Dataproc Operator parameters (#15622)``
+
```
Comment (potiuk) [inline airflow/providers/google/CHANGELOG.rst]:
```suggestion
```

**Comment by potiuk** [comment]
I removed Changelog entry. I am going to describe it soon but changelog entries (unless we introduce bigger description for breaking changes) should be managed by release manager (semi-automatically), not the author.

**Comment by eladkal** [comment]
@Daniel-Han-Yang can you fix the Pylint test?

**Comment by potiuk** [comment]
and rebase to latest `main` please :)

**Comment by turbaszek** [comment]
> > It looks like a breaking change. Can we avoid it and keep backward compatibility with warning? This change also requires an entry in changelog to facilitate migration. See: https://github.com/apache/airflow/blob/master/airflow/providers/google/CHANGELOG.rst
> 
> Thank you so much for doing this change that I haven't had time to do! +1 to @mik-laj 's comment about it being a breaking change.

@leahecole should we make similar change in other operators? AFAIK we assumed to use location everywhere to make the interface consistent.

**Comment by leahecole** [comment]
I just reviewed a bunch of the other GCP operators and it seems like most of the APIs use `location`, but there are a few outliers that do not, including our friend Dataproc here. I think it's fine to leave the other operators alone but I'll be sure to open issues if I notice other inconsistencies. (I'm a little salty that it's not consistent across and may file some issues internally too 😁 )

**Comment by Daniel-Han-Yang** [comment]
@potiuk @eladkal  I've just rebased on the PR again as it seems that there was a conflict. is there anything else I'll need to do on my side? Appreciate your guidance on this.

**Pull request merged**